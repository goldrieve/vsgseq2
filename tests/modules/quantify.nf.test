nextflow_process {

    name "Test Process QUANTIFY"
    script "modules/quantify.nf"
    process "QUANTIFY"

    test("Should run salmon quantification") {

        when {
            params {
                cores = 1
            }
            process {
                """
                // tuple val(meta), path(trimmed)
                input[0] = [ [ id: 'late_se', single_end: true ], file("${projectDir}/data/reads/latesingle.fq.gz") ]
                input[1] = params.cores
                input[2] = "${projectDir}/tests/data/salmon_index"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
