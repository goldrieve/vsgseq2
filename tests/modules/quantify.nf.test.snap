{
    "Should run salmon quantification": {
        "content": [
            "quant.sf:md5,a20edaff25b16ad270dcb7ad60b3aab2"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-30T11:16:38.311653143"
    }
}